app/actions/remote/channel.ts

```diff
@@ -557,6 +558,74 @@ export const switchToChannelByName = async (serverUrl: string, channelName: stri
     }
 };
 
+export async function getChannelMemberCountsByGroup(serverUrl: string, channelId: string, includeTimezones: boolean) {
+    try {
+        const client = NetworkManager.getClient(serverUrl);
+        const channelMemberCountsByGroup = await client.getChannelMemberCountsByGroup(channelId, includeTimezones);
+        return {channelMemberCountsByGroup};
+    } catch (error) {
+        return {error};
+    }
+}
+
+export async function getChannelTimezones(serverUrl: string, channelId: string) {
+    try {
+        const client = NetworkManager.getClient(serverUrl);
+        const channelTimezones = await client.getChannelTimezones(channelId);
+        return {channelTimezones};
+    } catch (error) {
+        return {error};
+    }
+}
+
+export async function getOrCreateDirectChannel(serverUrl: string, otherUserId: string, shouldSwitchToChannel = true) {
+    const operator = DatabaseManager.serverDatabases[serverUrl]?.operator;
+    if (!operator) {
+        return {error: `${serverUrl} database not found`};
+    }
+
+    const currentUserId = await queryCurrentUserId(operator.database);
+    const channelName = getDirectChannelName(currentUserId, otherUserId);
+
+    const channel = await queryChannelByName(operator.database, channelName);
+    let result;
+    if (channel) {
+        result = {channel};
+    } else {
+        try {
+            const client = NetworkManager.getClient(serverUrl);
```

Comment:
I prefer if we extract the client into its own try-catch so as to catch the error specifically for the Network Manager as compared to the errors that might be thrown from the rest api calls.